SeafarersAPI/Data/Repositories/IRepository.cs

```diff
@@ -0,0 +1,12 @@
+﻿namespace SeafarersAPI.Data.Repositories
+{
+    using System.Collections.Generic;
+
+    public interface IRepository<T>
```

Comment:
Added an interface that can be shared amongst all repositories. I put the basics but we might want to add more later.

Reply:
Another good point.